src/dataset/impl_targets.rs

```diff
@@ -1,22 +1,54 @@
 use std::collections::HashMap;
 
 use super::{
-    AsProbabilities, AsTargets, AsTargetsMut, CountedTargets, DatasetBase, FromTargetArray, Label,
-    Labels, Pr, Records,
+    AsMultiTargets, AsMultiTargetsMut, AsProbabilities, AsSingleTargets, AsSingleTargetsMut,
+    CountedTargets, DatasetBase, FromTargetArray, Label, Labels, Pr, Records,
 };
 use ndarray::{
     Array1, Array2, ArrayBase, ArrayView2, ArrayViewMut2, Axis, CowArray, Data, DataMut, Dimension,
     Ix1, Ix2, Ix3, OwnedRepr, ViewRepr,
 };
 
-impl<'a, L, S: Data<Elem = L>> AsTargets for ArrayBase<S, Ix1> {
+impl<'a, L, S: Data<Elem = L>> AsSingleTargets for ArrayBase<S, Ix1> {
     type Elem = L;
 
     fn as_multi_targets(&self) -> ArrayView2<L> {
         self.view().insert_axis(Axis(1))
     }
 }
 
+impl<'a, L, S: Data<Elem = L>> AsSingleTargets for ArrayBase<S, Ix2> {
+    type Elem = L;
+
+    fn as_multi_targets(&self) -> ArrayView2<L> {
+        self.view()
+    }
+}
+
+impl<'a, L, S: Data<Elem = L>> AsMultiTargets for ArrayBase<S, Ix1> {
+    type Elem = L;
+
+    fn as_multi_targets(&self) -> ArrayView2<L> {
+        self.view().insert_axis(Axis(1))
+    }
+
+    fn ntargets(&self) -> usize {
+        1
+    }
+}
```

Comment:
I don't know if this is possible, but maybe you can do a blanket impl like `impl AsMultiTargets for T where T: AsSingleTargets` instead.

Reply:
Actually, scratch that. We should make `AsMultiTargets` as supertrait of `AsSingleTarget`. Maybe `AsMultiTargets` should have its name changed back to `AsTargets`, since it encapsulates both single and multi-target data.

Reply:
Indeed, I think making `AsMultiTargets` a supertrait of `AsSingleTarget` would be helpful. I would opt for a clearer naming however, by keeping `AsSingleTarget` and `AsMultiTarget`. We should just precise in the doc that `AsSingleTarget` is simply an edge case of `AsMultiTarget` with `ntarget = 1`.

Reply:
That makes sense